source_code/src/AES/aes.c

```diff
@@ -22,10 +22,13 @@
 #define F(x)   (((x)<<1) ^ ((((x)>>7) & 1) * 0x1b))
 #define FD(x)  (((x) >> 1) ^ (((x) & 1) ? 0x8d : 0))
 
-#define BACK_TO_TABLES
 #ifdef BACK_TO_TABLES
 
-const uint8_t sbox[256] __attribute__ ((__progmem__)) = {		// forward s-box
+#if defined(__AVR__)
+const uint8_t sbox[256] __attribute__ ((__progmem__)) = {        // forward s-box
+#else
+const uint8_t sbox[256] = {        // forward s-box

```

Comment:
I don't get the question.

I assume you are only talking about the AVR implementation even though you marked a different line.

Whatever you asked: You still need the progmem here, as it is the (old) code that is used by the current firmware.

If you use startup tables, you do not even reach this line, because `BACK_TO_TABLES` needs to be undefined first.